compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEUtils.cpp

```diff
@@ -123,7 +122,13 @@ FailureOr<std::array<uint32_t, 3>> getAIEMatmulInstructionSize(Type elTypeLhs,
 
   if (allFloatingPoint) {
     if (nBitsLhs == 16 && nBitsRhs == 16 && nBitsAcc == 32) {
-      return std::array<uint32_t, 3>{4, 4, 8};
+      if (targetDevice == AMDAIEDevice::npu4) {
```

Comment:
The integer sizes specific to `npu4` is not known to me and I tried looking into https://github.com/Xilinx/mlir-aie/blob/main/include/aie/Dialect/AIEVec/IR/AIEVecOps.td but I believe since npu4 hasn't been made public into `mlir-aie` yet, this will need to wait.

I've added an assert - let me know if I should remove it.

Also, I've now moved the API to AMDAIEDeviceModel class in `iree_aie_runtime.cc`.

Reply:
> The integer sizes specific to npu4 is not known to me 

i32 cases shouldn't matter, anything should work as it's scalarized.
i8 is `8x8x8` as well.
Note bfloat16 is only 8x8x8 here as bfp16 is used under the hood for the ukernel we're targeting

Reply:
Yeah, so I see that we maintain a [map of intrinsics for all integer types](https://github.com/nod-ai/iree-amd-aie/blob/fc0480187f9dabb2cbf077a3ed643a8fd5cf6fca/runtime/src/iree-amd-aie/aie_runtime/iree_aie_runtime.cc#L674-L700) - it is those specific instrinsic sizes for all types that I'm not sure of. currently for `npu4`. How to derive the same (and what is the source where I can take a look at for `npu4`) ?

One thing which I thought was to just change 4 -> 8 and 2 -> 4 and have a new API `getNpu4IntegerMatmulInstructionSize`.

I could add just an `i8` instance for `npu4` but that'd be too specific for the integer types as currently we do that only for float types whereas we have a generic intrinsic map for integer bit-widths for `npu1_4col`.

Hence currently added an assert.

Reply:
Yeah, I don't think we should add them in this PR. We should throw an error though instead of an assert

Reply:
Done.